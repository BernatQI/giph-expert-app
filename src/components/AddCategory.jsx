import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState('');

  const onInputChange = ({target}) => {
    setInputValue(target.value);
  }

  const onSubmit = event => {
    event.preventDefault();
    const value = inputValue.trim();

    if(value.length <= 1 && setInputValue) return;

    onNewCategory(value);
    setInputValue('');
  }

  return(
    <form onSubmit={event => onSubmit(event)}>
      <input
        type='text'
        placeholder='Search Giphs...'
        value={inputValue}
        onChange={onInputChange} />
    </form>
  )
}