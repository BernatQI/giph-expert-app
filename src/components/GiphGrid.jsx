import { useEffect } from "react";
import { getGiphs } from "../helpers/getGiphs";

export const GiphGrid = ({category}) => {

  useEffect(() => {
    getGiphs(category);
  }, [])

  return(
    <article>
      <h3 className="GiphGrid-title">{category}</h3>
    </article>
  )
}