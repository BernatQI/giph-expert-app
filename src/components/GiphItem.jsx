import PropTypes from 'prop-types';

export const GiphItem = ({title, url}) => {

    return(
      <div className="card">
            <img className="giph-img" src={url} alt={title} />
          <h3>{`Title: ${title === '' ? 'No Title' : title}`}</h3>
        </div>
    )
}

GiphItem.propTypes = {
  title : PropTypes.string.isRequired,
  url   : PropTypes.string.isRequired
}