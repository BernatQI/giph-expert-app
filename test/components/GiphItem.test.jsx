import {render, screen} from '@testing-library/react'
import { GiphItem } from "../../src/components/GiphItem"

describe('Testing <GiphItem />', () => {

  const title = 'BernatQi';
  const url = 'https://bernatqi.dev/';

  test('Should match with snapshot', () => {
    const {container} = render(<GiphItem title={title} url={url} />);
    
    expect(container).toMatchSnapshot();
  });
  
  test('Shuld show proper image\Â´s url & alt', () => {
    render(<GiphItem title={title} url={url} />);
    //screen.debug();

    //expect(screen.getByRole('img').src).toBe(url);
    //expect(screen.getByRole('img').alt).toBe(title);
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('Sholod show title in component', () => {
    render(<GiphItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
})