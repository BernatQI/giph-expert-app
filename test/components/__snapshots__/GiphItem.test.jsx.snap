// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GiphItem /> Should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="BernatQi"
      class="giph-img"
      src="https://bernatqi.dev/"
    />
    <h3>
      Title: BernatQi
    </h3>
  </div>
</div>
`;
